iceoryx_dust/container/include/iox/detail/fixed_position_container.inl

```diff
@@ -51,6 +51,118 @@ inline FixedPositionContainer<T, CAPACITY>::~FixedPositionContainer() noexcept
     }
 }
 
+template <typename T, uint64_t CAPACITY>
+inline FixedPositionContainer<T, CAPACITY>::FixedPositionContainer(const FixedPositionContainer& rhs) noexcept
+{
+    *this = rhs;
+}
+
+template <typename T, uint64_t CAPACITY>
+inline FixedPositionContainer<T, CAPACITY>::FixedPositionContainer(FixedPositionContainer&& rhs) noexcept
+{
+    *this = std::move(rhs);
+}
+
+template <typename T, uint64_t CAPACITY>
+inline FixedPositionContainer<T, CAPACITY>&
+FixedPositionContainer<T, CAPACITY>::operator=(const FixedPositionContainer& rhs) noexcept
+{
+    if (this != &rhs)
+    {
+        IndexType i = 0U;
+        auto lhsSize = size();
+        auto minSize = algorithm::minVal(lhsSize, rhs.size());
+
+        // copy using copy assignment
+        for (; i < minSize; ++i)
+        {
+            m_data[i] = rhs.m_data[i];
+            m_status[i] = rhs.m_status[i];
+            m_next[i] = rhs.m_next[i];
```

Comment:
This is not quite correct. You could have the following situation on either of the two containers
``` 
+---+---+---+---+---+---+ - +
| 0 | 1 | 2 | 3 | 4 | 5 | I |   <- index
+---+---+---+---+---+---+ - +
| F | F | U | U | U | F |   |   <- status (F = free; U = used)
+---+---+---+---+---+---+ - +
| 1 | 5 | 3 | 4 | I | I |   |   <- next index with same status
+---+---+---+---+---+---+ - +
```

I think you need to iterate over both container and use the iterator for the copy operation. With this you even don't need to change the status and next index.

Another option would be to iterate over `rhs` and fill `this` via the indices but then you have to distinguish if the slot is free or used.

The latter one might result in a slightly better performance since all used slots will be consecutive.

Reply:
Apologies, but I am still somewhat unclear.

In the first scenario, I don't quite comprehend why there is no need to modify the `status` and `next` when using iterators to execute the copy operation. The dereferenced iterator `(*iter)` returns a reference to `m_data[i]`. If the value of `m_data` is updated via an assignment operation, it appears that the values for `status` and `next index` won't get updated. Or are you suggesting that these values get updated automatically through the use of functions like `emplace()`?

For the second scenario, I am not quite sure about the meaning of "all used slots will be consecutive".

```
// this
+---+---+---+---+---+---+ - +
| 0 | 1 | 2 | 3 | 4 | 5 | I |   <- index
+---+---+---+---+---+---+ - +
| F | F | U | U | U | F |   |   <- status (F = free; U = used)
+---+---+---+---+---+---+ - +
| 1 | 5 | 3 | 4 | I | I |   |   <- next index with same status
+---+---+---+---+---+---+ - +

// rhs
+---+---+---+---+---+---+ - +
| 0 | 1 | 2 | 3 | 4 | 5 | I |   <- index
+---+---+---+---+---+---+ - +
| F | U | F | U | F | F |   |   <- status (F = free; U = used)
+---+---+---+---+---+---+ - +
| 2 | 3 | 4 | I | 4 | I |   |   <- next index with same status
+---+---+---+---+---+---+ - +
```
Do you mean `this` will become following after copy?

```
// new this
+---+---+---+---+---+---+ - +
| 0 | 1 | 2 | 3 | 4 | 5 | I |   <- index
+---+---+---+---+---+---+ - +
| U | U | F | F | F | F |   |   <- status (F = free; U = used)
+---+---+---+---+---+---+ - +
| 1 | I | 3 | 4 | 5 | I |   |   <- next index with same status
+---+---+---+---+---+---+ - +
```

Also, if you think any document that I missed, please tell me. Thank you.

Reply:
No need to apologize :)

If you use the iterator on the source container you would iterate through all slots which contain valid data and therefore you could use the copy assignment operator on the dereferenced iterator, then increment the iterator and continue as long as it is not the end iterator. This will only change the `data` and and `status` and `next index` are already pointing to a valid used location. When the end iterator is reached, you would need to continue with `emplace` and if the destination container had more elements than the source container `erase` must be used on the remaining used slots.

Regarding the second option. Yes, that's exactly what I meant. I guess this option would result in less complex code, especially if `clear` would be called on the destination container before any of the other operations.

I think you now know about all the relevant documents for this PR :)

Reply:
Appreciate for your clear reply~